framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1177,6 +1185,11 @@ public void addAnnotation(AnnotationMirror a) {
          *     excluding the receiver
          */
         /*package-private*/ void setParameterTypes(List<AnnotatedTypeMirror> params) {
+            // Record varargsType when initializing paramTypes
+            if (isVarArgs() && paramTypes == null) {
```

Comment:
I don't think this works for the use in `shallowCopy` - there, shouldn't you also call `setVarargType`, to make sure that field is properly set?
I'm also not sure the combination of viewpoint adaptation and varargs works for this, but I'm fine with adding an issue and test for that separately.